samza-core/src/main/java/org/apache/samza/container/RunLoop.java

```diff
@@ -276,6 +276,9 @@ private void runTasks(IncomingMessageEnvelope envelope) {
           consumerMultiplexer.tryUpdate(envelope.getSystemStreamPartition(elasticityFactor));
```

Comment:
In the comments above 
"this condition happens when only a subset of keyBuckets of an SSP are being consumed at this container (but not all keyBuckets), therefore, we 
fake the consumption of the envelope, 
and decrement the metric, since this envelope's processing is  skipped.